README.md

```diff
@@ -34,6 +33,43 @@ Special notes:
  * Since we cannot distribute Java Richards, a patch is required to derive the
    tracing version (`patches/trace_java_richards.diff`)
 
+## Using the scripts here to analyse an existing benchmark
+
+### Requirements
+
+  * Python 2.7 (the code here is not Python 3.x ready)
+  * bzip2 / buzip2
+  * Python modules: numpy, rpy2
+  * R (e.g. `sudo apt-get install r-base`)
+  * Python modules required for plotting: matplotlib, seaborn
+  * Required for generating LaTeX tables: pdflatex (e.g. `sudo apt-get install tetex-live`)
+
+### CSV format
+
+Scripts here expect to work with benchmark data in CSV format. The first row
+should contain a header with a process execution id, benchmark name and sequence
+of iteration numbers. Subsequent rows should contain an id, benchmark name and
+times (in seconds) for each benchmark iteration, e.g.:
+
+```
+    process num, bench_name, 0, 1, 2, ...
+    0, spectral norm, 0.2, 0.1, 0.4, ...
+    1, spectral norm, 0.3, 0.15, 0.2, ...
+```
+
+### Usage
+
+The Python script `bin/warmup` should be used as a front-end to all other scripts.
+The script can be used to generate JSON containing summary statistics for
+the input data, PDF plots or LaTeX tables.
+
+The script also needs the names of the language and VM under test, and the
+output of `uname -a` on the machine the benchmarks were run on. Example usage:
+
+```
+./bin/warmup  --output-plots plots.pdf --output-json summary.json -l javascript -v V8 -u "Linux bencher2 3.16.0-4-amd64 #1 SMP Debian 3.16.36-1+deb8u2 (2016-10-19) x86_64 GNU/Linux" results.csv
```

Comment:
We can simplify this a bit by saying `` -u `uname -a` ``? It might make the command look a little less intimidating!